sycl/include/sycl/queue.hpp

```diff
@@ -3680,6 +3833,38 @@ class __SYCL_EXPORT queue : public detail::OwnerLessBase<queue> {
                                const detail::code_location &CodeLoc,
                                bool IsTopCodeLoc) const;
 
+#ifdef __INTEL_PREVIEW_BREAKING_CHANGES
+  event submit_direct_with_event_impl(
+      nd_range<1> Range, const detail::v1::SubmissionInfo &SubmitInfo,
+      const detail::v1::KernelRuntimeInfo &KRInfo,
+      const detail::code_location &CodeLoc, bool IsTopCodeLoc) const;
+
+  event submit_direct_with_event_impl(
+      nd_range<2> Range, const detail::v1::SubmissionInfo &SubmitInfo,
+      const detail::v1::KernelRuntimeInfo &KRInfo,
+      const detail::code_location &CodeLoc, bool IsTopCodeLoc) const;
+
+  event submit_direct_with_event_impl(
+      nd_range<3> Range, const detail::v1::SubmissionInfo &SubmitInfo,
+      const detail::v1::KernelRuntimeInfo &KRInfo,
+      const detail::code_location &CodeLoc, bool IsTopCodeLoc) const;
+
+  void submit_direct_without_event_impl(
+      nd_range<1> Range, const detail::v1::SubmissionInfo &SubmitInfo,
+      const detail::v1::KernelRuntimeInfo &KRInfo,
+      const detail::code_location &CodeLoc, bool IsTopCodeLoc) const;
+
+  void submit_direct_without_event_impl(
+      nd_range<2> Range, const detail::v1::SubmissionInfo &SubmitInfo,
+      const detail::v1::KernelRuntimeInfo &KRInfo,
+      const detail::code_location &CodeLoc, bool IsTopCodeLoc) const;
+
+  void submit_direct_without_event_impl(
+      nd_range<3> Range, const detail::v1::SubmissionInfo &SubmitInfo,
+      const detail::v1::KernelRuntimeInfo &KRInfo,
+      const detail::code_location &CodeLoc, bool IsTopCodeLoc) const;
+#endif //__INTEL_PREVIEW_BREAKING_CHANGES
```

Comment:
Can we keep new methods outside `class queue` as free functions? Once we pass ABI boundary, free functions can always do `getSyclObjImpl(sycl::queue)` and have access to all the internals.

That might save a bunch of `friend` declarations and possibly even one layer of tiny wrappers.

Reply:
Because that would be aligned with the public APIs. Old submission is member-function, new free-functions-enqueue is free functions. I think it would make sense if `*_impl` match the public API. That said, this is subjective, so if other folks don't see value in this suggestion, ignore it.